apps/backend/src/utils/nginx_parser.py

```
@@ -0,0 +1,553 @@
+            brace_count = content.count('{') - content.count('}')
+            if brace_count != 0:
+                validation['is_valid'] = False
+                validation['errors'].append(f"Unbalanced braces: {abs(brace_count)} {'opening' if brace_count > 0 else 'closing'} braces")
+            
+            # Check for missing semicolons
+            lines = content.split('\n')
+            for line_num, line in enumerate(lines, 1):
+                line = line.strip()
+                if line and not line.startswith('#') and not line.endswith((';', '{', '}')):
+                    # Skip block declarations
+                    if not any(block in line for block in ['server', 'location', 'upstream', 'if']):
```

Ruff (0.12.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)